src/components/Explorer/File.tsx

```diff
@@ -1,23 +1,26 @@
 "use client"
 
-import Link, { LinkProps } from "next/link";
-import { usePathname } from "next/navigation";
-import { ReactNode } from "react";
+import { ReactNode } from 'react';
+import Link, { LinkProps } from 'next/link';
+import { usePathname } from 'next/navigation';
+import { useTabs } from '@/hooks/useTabs';
 
 interface FileProps extends LinkProps {
-  children: ReactNode
+  children: ReactNode[];
 }
 
 export function File(props: FileProps) {
-  const pathName = usePathname()
+  const { addTab } = useTabs();
+  const pathName = usePathname();
 
-  const isCurrentActive = pathName === props.href
+  const isCurrentActive = pathName === props.href;
 
   return (
     <Link
       data-active={isCurrentActive}
-      className="flex text-sm items-center gap-2 py-1 px-4 pl-10 hover:bg-[#2a273f] hover:text-[#E0DEF2] data-[active=true]:bg-[#2a273f] data-[active=true]:text-[#E0DEF2]" 
-      {...props} 
+      onClick={() => addTab(props)}
```

Comment:
I was thinking about this line, what happens if the user opens a file by typing the file URL on the browser? It will not mark that file as open, right? So, maybe we should use an `useEffect` hook to detect when the `pathName` changes and mark that file/url as open, what do you think? 

This code will probably not be here then, we'll move that `useEffect` to inside the context provider.

Reply:
I had tried that, and a tried again after your suggestion, but the result came with a weird bug, whenever you tried to close a open file that was active, it keep opened, I thot that the effect it would only run once the path changes, but seems like it was triggering before the page redirect, the way I did it was starting the state with the pathname if exists, if there is another way to try, I accept any suggestions.